source/core/slang-platform.cpp

```diff
@@ -171,7 +171,7 @@ SLANG_COMPILE_TIME_ASSERT(E_OUTOFMEMORY == SLANG_E_OUT_OF_MEMORY);
 {
     handleOut = nullptr;
 
-	void* h = dlopen(platformFileName, RTLD_NOW | RTLD_LOCAL);
+	void* h = dlopen(platformFileName, RTLD_NOW | RTLD_GLOBAL);
```

Comment:
This is a subtle change to be making, and we should be careful about it.

The use of `RTLD_LOCAL` was intentional here because it means that the compiler can load something like `dxcompiler.so` (for a Unix-y build of dxc) from the specified path and *not* have to worry about other code in the same process that might also want to load dxc (but which might want a different version).

I guess the tricky bit is that for some of the stuff like `pthread` we really do want to have a global load, since we are actually loading it in order to influence what subsequently gets loaded.

I think we can let this change in for right now, but it seems like we actually need/want to update the shared library loader interface to support some minimal flags (or a `loadGlobal` sort of entry point), so that we can specify which of the two modes we are looking for.

Reply:
Agreed, we should change the interface of dll loader to support the flag.